pasqal_cloud/client.py

```diff
@@ -108,19 +109,42 @@ def _request(
         payload: Optional[Union[Mapping, Sequence[Mapping]]] = None,
         params: Optional[Mapping[str, Any]] = None,
     ) -> JSendPayload:
-        rsp = requests.request(
-            method,
-            url,
-            json=payload,
-            timeout=TIMEOUT,
-            headers={"content-type": "application/json"},
-            auth=self.authenticator,
-            params=params,
+        successful_request: bool = False
+        iteration: int = 0
+        while iteration <= HTTP_RETRIES and not successful_request:
+            # time = (interval seconds * exponent rule) ^ retries
+            delay = (1 * 2) ** iteration
+            rsp = requests.request(
+                method,
+                url,
+                json=payload,
+                timeout=TIMEOUT,
+                headers={"content-type": "application/json"},
+                auth=self.authenticator,
+                params=params,
+            )
+            try:
+                rsp.raise_for_status()
+                successful_request = True
+            except Exception as e:
+                if (
+                    rsp.status_code not in {408, 425, 429, 500, 502, 503, 504}
+                    or iteration == HTTP_RETRIES
+                ):
+                    raise e
+
+            if successful_request:
+                data: JSendPayload = rsp.json()
+                return data
+
+            time.sleep(delay)
```

Comment:
Should we log to the user that the request failed and we are retrying?

Reply:
I'm not a fan of adding logs to SDKs, they should be silent in the run time

Reply:
Yeah they should be silent by default I agree. My recommendation would be to add those log statement but configure the logger so that they do not show up unless the user changes the logger configuration explicitly in his script. Anyway this is outside the scope of this MR, so I'll resolve